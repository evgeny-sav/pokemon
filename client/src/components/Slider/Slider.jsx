import React from 'react';
import { shallowEqual, useSelector } from 'react-redux';
import { Carousel, Image } from 'react-bootstrap';

const Slider = () => {
  const { sprites } = useSelector(state => state.pokemon.data, shallowEqual);
  return (
    <Carousel
      className="text-center"
      nextIcon={
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="#FFCB05"
          width="20"
          height="20"
          viewBox="0 0 8 8"
        >
          <path d="M2.75 0l-1.5 1.5L3.75 4l-2.5 2.5L2.75 8l4-4-4-4z" />
        </svg>
      }
      prevIcon={
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="#FFCB05"
          width="20"
          height="20"
          viewBox="0 0 8 8"
        >
          <path d="M5.25 0l-4 4 4 4 1.5-1.5L4.25 4l2.5-2.5L5.25 0z" />
        </svg>
      }
    >
      {Object.keys(sprites)
        .sort(a => (/front_default/.test(a) ? -1 : 1))
        .reduce((acc, key) => {
          const imgUrl = sprites[key];
          if (imgUrl !== null) {
            const slide = (
              <Carousel.Item key={`${name}${Math.random()}`}>
                <Image width={200} src={imgUrl} alt={name} />
              </Carousel.Item>
            );
            acc.push(slide);
          }
          return acc;
        }, [])}
    </Carousel>
  );
};

export default Slider;
