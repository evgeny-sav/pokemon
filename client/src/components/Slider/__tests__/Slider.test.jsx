import React from 'react';
import { create } from 'react-test-renderer';
import Slider from '../Slider';

jest.mock('react-redux', () => ({
  useSelector: () => ({
    sprites: {},
  }),
}));

describe('Slider', () => {
  it('should render', () => {
    const component = create(<Slider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
