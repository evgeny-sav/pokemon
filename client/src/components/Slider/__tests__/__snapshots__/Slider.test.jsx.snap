// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render 1`] = `
<div
  className="text-center carousel slide"
  onKeyDown={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
>
  <ol
    className="carousel-indicators"
  />
  <div
    className="carousel-inner"
  />
  <a
    className="carousel-control-prev"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    <svg
      fill="#FFCB05"
      height="20"
      viewBox="0 0 8 8"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.25 0l-4 4 4 4 1.5-1.5L4.25 4l2.5-2.5L5.25 0z"
      />
    </svg>
    <span
      className="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    className="carousel-control-next"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    <svg
      fill="#FFCB05"
      height="20"
      viewBox="0 0 8 8"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.75 0l-1.5 1.5L3.75 4l-2.5 2.5L2.75 8l4-4-4-4z"
      />
    </svg>
    <span
      className="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;
